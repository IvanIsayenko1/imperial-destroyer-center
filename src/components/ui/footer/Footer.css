.footer {
  margin-top: auto;
  padding: 1rem;
  border-top: 1px solid var(--sith-red);
}

.footer-content {
  color: var(--rebel-base);
  font-size: 0.75rem;
  text-align: center;
  text-transform: uppercase;
  letter-spacing: 2px;
  opacity: 0.7;
  transition: opacity 0.3s ease;
}

.footer-content:hover {
  opacity: 1;
}

.footer-content a {
  color: var(--sith-red);
  text-decoration: none;
}

.footer-content a:hover {
  text-shadow: 0 0 8px var(--sith-red);
}
