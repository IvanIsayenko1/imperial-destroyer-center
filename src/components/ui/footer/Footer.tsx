import React from "react";
import "./Footer.css";

export const Footer: React.FC = () => {
  return (
    <footer className="footer">
      <div className="footer-content">
        Â© {new Date().getFullYear()} Imperial Fleet Database
      </div>
    </footer>
  );
};
