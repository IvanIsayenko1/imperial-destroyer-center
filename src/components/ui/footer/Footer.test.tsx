import { render, screen } from "@testing-library/react";
import { Footer } from "./Footer";

describe("Footer component", () => {
  it("should render footer content with the current year", () => {
    render(<Footer />);
    const footerContent = screen.getByText(/Imperial Fleet Database/i);
    const currentYear = new Date().getFullYear();
    expect(footerContent).toHaveTextContent(
      `Â© ${currentYear} Imperial Fleet Database`
    );
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
