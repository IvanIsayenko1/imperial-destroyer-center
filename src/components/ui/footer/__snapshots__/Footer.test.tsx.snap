// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-content"
    >
      Â© 2024 Imperial Fleet Database
    </div>
  </footer>
</DocumentFragment>
`;
